receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -442,56 +437,63 @@ func (jr *jReceiver) buildProcessor(address string, factory apacheThrift.TProtoc
 }
 
 func (jr *jReceiver) startCollector(host receiver.Host) error {
-	tch, terr := tchannel.NewChannel("jaeger-collector", new(tchannel.ChannelOptions))
-	if terr != nil {
-		return fmt.Errorf("failed to create NewTChannel: %v", terr)
+	if !jr.collectorGRPCEnabled() && !jr.collectorHTTPEnabled() && !jr.collectorThriftEnabled() {
+		return nil
```

Comment:
Shouldn't this return error since nothing is enabled?

Reply:
In `StartTraceReception()` both the collector and agent protocols are started.  A receiver could be configured for only agent protocols which would be a valid configuration but still return nil here.

https://github.com/open-telemetry/opentelemetry-collector/blob/39b9ea98ccc9b2e8cc547fa434ece6793233edf4/receiver/jaegerreceiver/trace_receiver.go#L212-L220

I had considered breaking each protocol into it's own "start" function instead of `startAgent()` and `startCollector()`, but I thought that might be too much churn for this PR.